### PR TITLE
6723: Error creating circle annotation

### PR DESCRIPTION
**Pull request by MuhweziDeo**
## Description

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix

## 6723

**What is the current behavior?**
MapStore crashes before having the chance to enter a radius while entering Lat and Lng

#6723

**What is the new behavior?**
It should be possible to enter coordinates before entering a radius without throwing an error.

#### how the issue was resolved
I found that the package https://www.npmjs.com/package/@turf/turf causes the app to crash at this line https://github.com/geosolutions-it/MapStore2/blob/master/web/client/reducers/annotations.js#L158. It's my guess that `circle` needs the radius. Without radius it causes the app to crash. So wrapping it a `try-catch solves the problem

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by MV88 — CHANGES_REQUESTED** [review]
Hi @MuhweziDeo 

is there a way to avoid the problem instead doing this try catch just to prevent the app from crashing. also in the catch you are not doing anything and the console is not helping the user.

can we use a very little value for radius if not defined so even it it is drawn then it is not visible
please make a few tries and let me know

i would

**Comment by MuhweziDeo** [comment]
> Hi @MuhweziDeo
> 
> is there a way to avoid the problem instead doing this try catch just to prevent the app from crashing. also in the catch you are not doing anything and the console is not helping the user.
> 
> can we use a very little value for radius if not defined so even it it is drawn then it is not visible
> please make a few tries and let me know
> 
> i would

Okay let try that thanks

File: web/client/reducers/annotations.js [hunk 1]
```diff
@@ -155,9 +155,11 @@ function annotations(state = {validationErrors: {}}, action) {
             if (validateCoordsArray(selected.properties.center)) {
                 center = selected.properties.center;
                 // turf/circle by default use km unit hence we divide by 1000 the radius(in meters)
+                // this try catch prevents the app from crashing when there is no radius maybe we can use a default value for radius incase action.radius === undefined
```
Comment (MV88) [inline web/client/reducers/annotations.js:158]:
```suggestion

```

**Review by MV88 — CHANGES_REQUESTED** [review]
can you add a unit test?

File: web/client/reducers/annotations.js (hunk 1 above)
Comment (MuhweziDeo) [reply web/client/reducers/annotations.js:158]: okay

**Review by MV88 — APPROVED** [review]

**Comment by MV88** [comment]
@ElenaGallo can you please test this in DEV?

**Comment by ElenaGallo** [comment]
@MuhweziDeo test done, please backport this PR to the stable branch. Thank you

**Pull request merged**